# AnalystLab Africa - Task 1: Dataset Understanding
## E-commerce Transactions Datset

Author: Brandon Mkhululi Muchenje

Date: 05 July 2026

In [5]:
#Importing the pandas libarary
import pandas as pd
import numpy as np

#Printing the outcome
print('Pandas successfully loaded')

import os


Pandas successfully loaded


In [9]:
#Loading the dataset
df = pd.read_csv("OnlineRetail.csv", encoding='ISO-8859-1')

#Print upon successful completion
print('E-commerce dataset has been loaded')

E-commerce dataset has been loaded


In [11]:
#Displaying the first few rows
df.head(10)

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/2010 8:26,2.55,17850.0,United Kingdom
1,536365,71053,WHITE METAL LANTERN,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/2010 8:26,2.75,17850.0,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
5,536365,22752,SET 7 BABUSHKA NESTING BOXES,2,12/1/2010 8:26,7.65,17850.0,United Kingdom
6,536365,21730,GLASS STAR FROSTED T-LIGHT HOLDER,6,12/1/2010 8:26,4.25,17850.0,United Kingdom
7,536366,22633,HAND WARMER UNION JACK,6,12/1/2010 8:28,1.85,17850.0,United Kingdom
8,536366,22632,HAND WARMER RED POLKA DOT,6,12/1/2010 8:28,1.85,17850.0,United Kingdom
9,536367,84879,ASSORTED COLOUR BIRD ORNAMENT,32,12/1/2010 8:34,1.69,13047.0,United Kingdom


In [17]:
#Displaying the number of rows and columns
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  object 
 1   StockCode    541909 non-null  object 
 2   Description  540455 non-null  object 
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  object 
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 33.1+ MB


In [23]:
#Displaying the datatypes of each column
df.dtypes

InvoiceNo       object
StockCode       object
Description     object
Quantity         int64
InvoiceDate     object
UnitPrice      float64
CustomerID     float64
Country         object
dtype: object

In [24]:
#Displaying Numerical Features
df.select_dtypes(include='number').columns

Index(['Quantity', 'UnitPrice', 'CustomerID'], dtype='object')

In [25]:
#Displaying Categorical Features
df.select_dtypes(include='object').columns

Index(['InvoiceNo', 'StockCode', 'Description', 'InvoiceDate', 'Country'], dtype='object')

In [29]:
#Assessing possible unique identifiers
for col in df.columns:
    if df[col].nunique() == len(df):
        print(col)

df.duplicated(subset=['InvoiceNo', 'StockCode']).sum()

np.int64(10684)

## Description of the Dataset
The dataset contains online e-commerce data transactions extracted from a real-world dataset. 
    
It contains 541909 rows and 8 columns, namely, details such as the invoice number, stock code, description, quantity, invoice date, 
unit price, customer id and country.
    
From those 8 columns, there is only 1 integer column, the quantity column, which makes sense as you cannot purchase 1.5 or 0.5 of a product. 
This is also one of the numerical features.
    
There are 2 float columns, which represent the other numerical features (UnitPrice and CustomerID), 
and the remaining are considered categorical features but the true categorical feature would be 'Country'.
CustomerID can't really be considered a float as you can only have a solid number for customer ID, suggesting further future cleaning or assessment
must be considered when assessing the column.
    
The 'InvoiceDate', 'InvoiceNo', 'StockCode' columns would need to be converted to the correct formats upon cleaning.
Description is free text and would not need converting.

Although CustomerID and InvoiceNo would typically serve as primary keys in a normalised relational schema (e.g., separate Customers and Invoices tables), in this flat transactional dataset they function as foreign keys, since each value repeats across multiple rows. No column in this dataset, as structured, qualifies as a true unique identifier on its own.